tests/parameter_space/test_parameters.py

```diff
@@ -0,0 +1,361 @@
+import sys
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+
+import numpy as np
+import pytest
+from CADETProcess.parameter_space.parameters import (
+    ChoiceParameter,
+    LinearConstraint,
+    LinearEqualityConstraint,
+    ParameterDependency,
+    ParameterDotPathSetter,
+    ParameterSpace,
+    RangedParameter,
+)
+from numpy.typing import NDArray
+
+sys.path.insert(0, str(Path(__file__).resolve().parents[2]))
+
+
+@dataclass
+class ObjWithDict:
+    """Object that contains a dict so we can test object → dict traversal."""
+    store: dict
+
+
+@dataclass
+class ObjLeaf:
+    """Object whose attribute will act as the leaf value."""
+    ka: float = 1e-5
+
+
+@pytest.fixture
+def dict_to_obj():
+    """Root dict ⟶ object.attr leaf."""
+    column = ObjLeaf(ka=1e-5)
+    return {"column": column}
+
+
+@pytest.fixture
+def obj_to_dict():
+    """Root object ⟶ dict['ka'] leaf."""
+    obj = ObjWithDict(store={"ka": 2e-5})
+    return obj
+
+
+def test_dict_to_object_traversal(dict_to_obj):
+    setter = ParameterDotPathSetter(
+        [dict_to_obj],
+        "column.ka"     # dict key → object attribute
+    )
+    new_val = 3.3e-4
+    setter.set_value(new_val)
+    assert dict_to_obj["column"].ka == new_val
+
+
+def test_object_to_dict_traversal(obj_to_dict):
+    setter = ParameterDotPathSetter(
+        [obj_to_dict],
+        "store.ka"      # object attribute → dict key
+    )
+    new_val = 4.4e-4
+    setter.set_value(new_val)
+    assert obj_to_dict.store["ka"] == new_val
+
+
+def test_missing_path_raises(dict_to_obj):
+    """No auto-creation: missing hop should raise KeyError or AttributeError."""
+    setter = ParameterDotPathSetter([dict_to_obj], "column.missing.leaf")
+    with pytest.raises(AttributeError):
+        setter.set_value(0.0)
+
+
+@pytest.fixture
+def random_dict():
```

Comment:
It's not really random, right? Try to be more specific, e.g. call it "nested_dict".